apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/ZipImpl.java

```diff
@@ -0,0 +1,216 @@
+            throw new IllegalArgumentException("'outputFolder' value is empty");
+        }
+        log.trace("Extracting file '{}' from outputFolder '{}'", zipFile, outputFolder);
+        boolean res = true;
+        byte[] buffer = new byte[BUF_SIZE];
+
+        try {
+
+            //create output directory is not exists
+            File folder = new File(outputFolder);
+            if (!folder.exists()) {
+                folder.mkdirs();
+            }
+
+            //get the zipped file list entry
+            try ( //get the zip file content
+                ZipInputStream zis = new ZipInputStream(new FileInputStream(zipFile))) {
+                //get the zipped file list entry
+                ZipEntry ze = zis.getNextEntry();
+
+                while (ze != null) {
+
+                    String fileName = ze.getName();
+                    File newFile = new File(outputFolder + File.separator + fileName);
+
+                    //create all non exists folders
+                    //else you will hit FileNotFoundException for compressed folder
+                    new File(newFile.getParent()).mkdirs();
+
+                    if (ze.isDirectory()) {
+                        newFile.mkdirs();
+                    } else {
+                        try (FileOutputStream fos = new FileOutputStream(newFile)) {
+                            int len;
+                            while ((len = zis.read(buffer)) > 0) {
+                                fos.write(buffer, 0, len);
+                            }
+                        }
+                    }
+                    ze = zis.getNextEntry();
+                }
+
+                zis.closeEntry();
+            }
+
+        } catch (IOException ex) {
+            log.error("Error extractiong zip file: {}", zipFile, ex);
+            res = false;
+        }
+        return res;
+    }
+    
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public byte[] compress(String zipFile, String inputFolder, Long filesTimeFromEpoch,
+                            FilenameFilter filenameFilter) {
+        Objects.requireNonNull(zipFile, "zipFile is NULL");
+        Objects.requireNonNull(inputFolder, "inputFolder is NULL");
+        if (zipFile.isEmpty()) {
+            throw new IllegalArgumentException("'zipFile' value is empty");
+        }
+        if (inputFolder.isEmpty()) {
+            throw new IllegalArgumentException("'inputFolder' value is empty");
+        }
+        long start = System.currentTimeMillis();
+        log.trace("Creating file '{}' in folder '{}', filesTimestamp = {}", zipFile, inputFolder, filesTimeFromEpoch);
+        byte[] zipCrcHash;
+        File directory = new File(inputFolder);
+        if (filenameFilter == null) {
+            filenameFilter = DEFAULT_CSV_FILE_FILTER;
+        }
+        // get file(s) listing from folder by filter (no recursion for subfolders(s) !)
+        File[] fileList = directory.listFiles(filenameFilter);
```

Comment:
General zip implementation should also zip subfolders

Reply:
No need to zip sub-folder now, may be later when we have it.